aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -94,6 +94,32 @@ SparseTensor& mul_sparse_scalar_(SparseTensor& t, Scalar v) {
   return mul_out_sparse_scalar(t, t, v);
 }
 
+// --------------------------------------------------------------------
+// log1p(SparseTensor)
+// --------------------------------------------------------------------
+
+// TODO: add in-place variant
```

Comment:
Should this be `out-of-place variant`? 

Would that be clone, coalesce, and then in-place log1p on values?

Reply:
I guess by 'in-place' I meant `t.coalesce_().log1p_()`. But since we don't have `coalesce_()`, so I put it as TODO. Should I reword it a bit?

Reply:
Oh I see. Is out-of-place `log1p` on sparse already implemented? I'm asking since I only see inplace variants.

Reply:
I think `log1p` secretly uses `log1p_out` [here](https://github.com/pytorch/pytorch/blob/7a048cdcd77dfee7d65c7624831663036fff9449/aten/src/ATen/native/UnaryOps.cpp#L38-L95)

Reply:
I see. Thanks! :)